### PR TITLE
feat(AdaptiveCardsInputToggle): implement Input.Toggle component

### PR DESCRIPTION
**Pull request by patricia0817**
Task link: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-292453

### CODE REVIEW

File: src/components/adaptive-cards/InputToggle/InputToggle.jsx [hunk 1]
```diff
@@ -1,23 +1,65 @@
...
  * @param {object} props  React props passed to the component
  * @param {object} props.data  Active cards definition
+ * @param {string} [props.className]  Custom CSS class to apply
+ * @param {object} [props.style]  Inline style for the component
  * @returns {object} JSX of the component
  */
-export default function InputToggle({data}) {
+export default function InputToggle({data, className, style}) {
+  const [cssClasses, sc] = webexComponentClasses('adaptive-cards-input-toggle', className);
+  const [inputValue, setInputValue] = useState(data.value || 'false');
```
Comment (cipak) [inline src/components/adaptive-cards/InputToggle/InputToggle.jsx:18]: use getValue() from the adaptive card context

File: src/components/adaptive-cards/InputToggle/InputToggle.jsx [hunk 2]
```diff
@@ -8,16 +10,60 @@ import {acPropTypes, registerComponent} from '../Component/Component';
...
+  const [cssClasses, sc] = webexComponentClasses('adaptive-cards-input-toggle', className);
+  const {getValue, setInput} = useContext(AdaptiveCardContext);
+  const inputValue = getValue() || 'false';
+  const [checked, setChecked] = useState(Boolean(data.value) || false);
+  const handleInputChange = () => {
+    setChecked((prevValue) => !prevValue);
+  };
+
+  useEffect(() => (
+    setInput({
```
Comment (cipak) [inline src/components/adaptive-cards/InputToggle/InputToggle.jsx:27]: id should be listed here, otherwise this value can't be attached to something

File: src/components/adaptive-cards/InputToggle/InputToggle.jsx [hunk 3]
```diff
@@ -8,16 +10,60 @@ import {acPropTypes, registerComponent} from '../Component/Component';
...
+ * @param {string} [props.className]  Custom CSS class to apply
+ * @param {object} [props.style]  Inline style for the component
  * @returns {object} JSX of the component
  */
-export default function InputToggle({data}) {
+export default function InputToggle({data, className, style}) {
+  const [cssClasses, sc] = webexComponentClasses('adaptive-cards-input-toggle', className);
+  const {getValue, setInput} = useContext(AdaptiveCardContext);
+  const inputValue = getValue() || 'false';
+  const [checked, setChecked] = useState(Boolean(data.value) || false);
```
Comment (cipak) [inline src/components/adaptive-cards/InputToggle/InputToggle.jsx:21]: no state here, use setValue from context, otherwise the validation and submit may work incorrectly

File: src/components/adaptive-cards/AdaptiveCard/AdaptiveCard.stories.js [hunk 4]
```diff
@@ -2169,6 +2169,31 @@ const exampleInputChoiceSet = {
...
+const exampleInputToggle = {
+  $schema: 'http://adaptivecards.io/schemas/adaptive-card.json',
+  type: 'AdaptiveCard',
+  version: '1.0',
+  body: [
+    {
+      type: 'Input.Toggle',
+      id: 'acceptTerms',
+      title: 'I accept the terms and agreements',
+      value: 'true',
```
Comment (cipak) [inline src/components/adaptive-cards/AdaptiveCard/AdaptiveCard.stories.js:2181]: let's demonstrate the full functionality of these fields:
``js
value: 'rejected',
valueOn: 'accepted',
valueOff: 'rejected',

File: src/components/adaptive-cards/InputToggle/InputToggle.jsx [hunk 5]
```diff
@@ -32,6 +80,8 @@ InputToggle.acPropTypes = {
   title: acPropTypes.title,
   type: acPropTypes.type,
   value: acPropTypes.value,
+  valueOff: acPropTypes.valueOff,
```
Comment (cipak) [inline src/components/adaptive-cards/InputToggle/InputToggle.jsx:83]: please add defaults for valueOn and valueOff

File: src/components/adaptive-cards/InputToggle/InputToggle.jsx [hunk 6]
```diff
@@ -8,16 +10,62 @@ import {acPropTypes, registerComponent} from '../Component/Component';
...
 }
 
 InputToggle.propTypes = {
   data: PropTypes.shape().isRequired,
+  className: PropTypes.string,
+  style: PropTypes.shape(),
+};
+
+InputToggle.defaultProps = {
+  className: '',
```
Comment (cipak) [inline src/components/adaptive-cards/InputToggle/InputToggle.jsx:50]: undefined

File: src/components/adaptive-cards/InputToggle/InputToggle.jsx (hunk 5 above)
Comment (cipak) [reply src/components/adaptive-cards/InputToggle/InputToggle.jsx:83]: and value, it seems

File: src/components/adaptive-cards/InputToggle/InputToggle.jsx [hunk 7]
```diff
@@ -8,16 +10,62 @@ import {acPropTypes, registerComponent} from '../Component/Component';
...
+ * @param {string} [props.className]  Custom CSS class to apply
+ * @param {object} [props.style]  Inline style for the component
  * @returns {object} JSX of the component
  */
-export default function InputToggle({data}) {
+export default function InputToggle({data, className, style}) {
+  const [cssClasses, sc] = webexComponentClasses('adaptive-cards-input-toggle', className);
+  const {value = 'false', valueOn = 'true', valueOff = 'false'} = data;
+  const {getValue, setInput, setValue} = useContext(AdaptiveCardContext);
+  const inputValue = getValue(data.id);
```
Comment (cipak) [inline src/components/adaptive-cards/InputToggle/InputToggle.jsx:21]: for consistency with other components, let's use the name `value` for the current value of the input (got from getValue()).

data.value can be called initialValue or can remain as data.value, as you prefer

File: src/components/adaptive-cards/Component/Component.jsx [hunk 8]
```diff
@@ -166,9 +169,19 @@ export default function Component({
...
       case acPropTypes.text:
       case acPropTypes.tooltip:
+      case acPropTypes.title:
       case acPropTypes.type:
       case acPropTypes.url:
+        break;
       case acPropTypes.value:
+      case acPropTypes.valueOff:
+      case acPropTypes.valueOn:
+        compData = {
```
Comment (cipak) [inline src/components/adaptive-cards/Component/Component.jsx:179]: why is this needed?

Comment (patricia0817) [reply src/components/adaptive-cards/Component/Component.jsx:179]: the linter doesn't allow us to directly mutate the values passed as parameters

**Review by cipak — APPROVED** [review]
